core/src/main/scala/org/apache/spark/internal/Logging.scala

```diff
@@ -126,11 +126,14 @@ trait Logging {
 
   private def initializeLogging(isInterpreter: Boolean, silent: Boolean): Unit = {
     if (Logging.isLog4j2()) {
-      // If Log4j is used but is not initialized, load a default properties file
-      val log4j2Initialized = !LogManager.getRootLogger
-        .asInstanceOf[org.apache.logging.log4j.core.Logger].getAppenders.isEmpty
+      val rootLogger = LogManager.getRootLogger.asInstanceOf[org.apache.logging.log4j.core.Logger]
+      // If Log4j 2 is used but is initialized by default configuration,
+      // load a default properties file
+      // (see org.apache.logging.log4j.core.config.DefaultConfiguration)
+      val needToInitializeLog4j2 = rootLogger.getAppenders.isEmpty ||
+        (rootLogger.getAppenders.size() == 1 && rootLogger.getLevel == Level.ERROR)
```

Comment:
This looks not reliable. It seems possible to wrongly treat a user log4j configuration as default configuration. Can we check if the configuration name is `DefaultConfiguration.DEFAULT_NAME`?

Reply:
https://github.com/apache/logging-log4j2/blob/5edf43758f104affad9842753e66be173fc4d68a/log4j-core/src/main/java/org/apache/logging/log4j/core/config/AbstractConfiguration.java#L708

Reply:
@viirya thanks for your comments. will copy that.

Reply:
> https://github.com/apache/logging-log4j2/blob/5edf43758f104affad9842753e66be173fc4d68a/log4j-core/src/main/java/org/apache/logging/log4j/core/config/AbstractConfiguration.java#L708

the name includes the "hash code", so it is hard to match the name absolutely. I check the class of `Configuration` instead.

Reply:
Hmm, I think we can compare with `DefaultConfiguration.DEFAULT_NAME + "@"`? Checking if it is `DefaultConfiguration` seems fine too.